core/src/main/scala/org/ensime/core/Analyzer.scala

```diff
@@ -169,43 +161,29 @@ class Analyzer(
   def startup: Receive = withLabel("startup") {
     case FullTypeCheckCompleteEvent =>
       reporter.enable()
-
-      // legacy clients expect to see AnalyzerReady and a
-      // FullTypeCheckCompleteEvent on connection.
-      //      broadcaster ! Broadcaster.Persist(AnalyzerReadyEvent)
-      //      broadcaster ! Broadcaster.Persist(FullTypeCheckCompleteEvent)
-      // Already handled by AnalyzerManager
       context.become(ready)
       unstashAll()
-
     case other =>
       stash()
   }
 
   def ready: Receive = withLabel("ready") {
     case FullTypeCheckCompleteEvent =>
       broadcaster ! FullTypeCheckCompleteEvent
-
     case RestartScalaCompilerReq(id, strategy) =>
-      restartCompiler(strategy, id)
-
+      restartCompiler(strategy)
     case SuspendAnalyzer =>
-      val now = clock.millis()
-      if (now - lastActivity >= LIFE) {
-        loadedFiles = scalaCompiler.loadedFiles // remember the state
-        scalaCompiler.askShutdown()
-        context.become(suspended)
-      }
-
+      loadedFiles = scalaCompiler.loadedFiles // remember the state
+      scalaCompiler.askShutdown()
+      context.become(suspended)
     case UnloadAllReq =>
-      restartCompiler(ReloadStrategy.UnloadAll, None)
+      restartCompiler(ReloadStrategy.UnloadAll)
       sender ! VoidResponse
-
     case TypecheckModule(id) =>
-      restartCompiler(ReloadStrategy.LoadProject, Some(id))
-      lastActivity = clock.millis()
+      countdown.cancel()
+      countdown = setCountdown()
```

Comment:
you don't need to do it here, look up to the method that calls `doAllTheThings`, you can put the logic in there or there might be a better way to run a side effect before running a partial function.